subsys/bluetooth/audio/ascs.c

```diff
@@ -897,18 +905,18 @@ static void ascs_ep_iso_disconnected(struct bt_bap_ep *ep, uint8_t reason)
 
 	if (ep->status.state == BT_BAP_EP_STATE_RELEASING) {
 		ascs_ep_set_state(ep, BT_BAP_EP_STATE_IDLE);
-	} else {
+	} else if (ep->status.state == BT_BAP_EP_STATE_STREAMING) {
 		/* The ASE state machine goes into different states from this operation
 		 * based on whether it is a source or a sink ASE.
 		 */
-		if (ep->status.state == BT_BAP_EP_STATE_STREAMING ||
-		    ep->status.state == BT_BAP_EP_STATE_ENABLING) {
-			if (ep->dir == BT_AUDIO_DIR_SOURCE) {
-				ascs_ep_set_state(ep, BT_BAP_EP_STATE_DISABLING);
-			} else {
-				ascs_ep_set_state(ep, BT_BAP_EP_STATE_QOS_CONFIGURED);
-			}
+		if (ep->dir == BT_AUDIO_DIR_SOURCE) {
+			ase->link_loss = true;
```

Comment:
Suggest to move this assignment next to `ep->reason = reason;`. While it is currently only used during the BT_BAP_EP_STATE_DISABLING handler, it may be useful to access at some later point, where it may as well reflect the actual value, which is also that `ase->ep->reason` is only valid if `ase->link_loss`. 

Alternatively, if you don't agree with the above, then the field in the struct should have a comment stating that this field is only valid for source ASEs in the streaming state.

Reply:
I was thinking about it for a while, and I think `link_loss` can only be set when ASE is in `BT_BAP_EP_STATE_STREAMING` or `BT_BAP_EP_STATE_DISABLING`.
Consider the because if set in e.g `BT_BAP_EP_STATE_ENABLING` and the server initiates Disable, we cannot go to QoS autonomously then. I updated the `test_cis_link_loss_in_enabling_state` test case to cover this scenario